pkg/apis/pipeline/v1alpha1/task_validation.go

```diff
@@ -26,12 +27,57 @@ import (
 )
 
 func (t *Task) Validate() *apis.FieldError {
+	if appliedConfig, present := t.Annotations["kubectl.kubernetes.io/last-applied-configuration"]; present {
```

Comment:
@dgageot so the general idea is that the `last-applied-configuration` annotation contains all of the fields, so we can use that to check if there are any extra fields :thinking: 

does it contain all the optional fields as well? id hazard a guess that it doesn't? (if it did, the annotations on a `Task` would contain all of the possible fields in a `Container` :thinking: )